buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy

```diff
@@ -129,50 +124,63 @@ class DistributedPerformanceTest extends PerformanceTest {
 
     @TaskAction
     void executeTests() {
-        createWorkerTestResultsTempDir()
         try {
             doExecuteTests()
         } finally {
             testEventsGenerator.release()
             generatePerformanceReport()
-            cleanTempFiles()
         }
     }
 
+    private File generateResultJson() {
+        File resultJson = File.createTempFile('distributedPerformanceTest', 'results.json')
+        resultJson.deleteOnExit()
```

Comment:
I guess the json is less than 100 KiB so it's fine to delete them on exit.

Reply:
This is not about the size of the `json`. We can also leave the JSON lying around the (project) `tmp` directory. Then it will be cleaned up by the next `clean` run.

I don't like to depend on the daemon shutting down to remove this file. The daemon lives to long (many builds) to make this feasible.

Reply:
Understood. Our daemon usually live through thousands of builds (or at least hundreds), I've done a cleanup action after the build.